bitex/interface/bitstamp.py

```diff
@@ -77,136 +77,129 @@ def bid(self, pair, price, size, *args, market=False, **kwargs):
-        payload = {'amount': size, 'price': price}
-        payload.update(kwargs)
-        if market:
-            return self.request('%s/market/%s/' % (side, pair), authenticate=True, params=payload)
-        return self.request('%s/%s/' % (side, pair), authenticate=True, params=payload)
+        kwargs.update({'amount': size, 'price': price})
+        suffix = 'market/' if market else ''
+        return self.request('%s/%s%s/' % (side, suffix, pair), authenticate=True, params=kwargs)
 
     @format_with(BitstampFormattedResponse)
     def order_status(self, order_id, *args, **kwargs):
         """Return the order status for the given order's ID."""
-        payload = {'id': order_id}
-        payload.update(kwargs)
-        return self.request('api/order_status/', authenticate=True, params=payload)
+        kwargs.update({'id': order_id})
+        return self.request('api/order_status/', authenticate=True, params=kwargs)
 
     @format_with(BitstampFormattedResponse)
     def open_orders(self, *args, pair=None, **kwargs):
         """Return all open orders."""
-        if pair:
-            return self.request('open_orders/%s/' % pair, authenticate=True, params=kwargs)
-        return self.request('open_orders/all/', authenticate=True, params=kwargs)
+        suffix = pair or 'all'
+        return self.request('open_orders/%s/' % suffix, authenticate=True, params=kwargs)
 
     @format_with(BitstampFormattedResponse)
     def cancel_order(self, *order_ids, **kwargs):
         """Cancel existing order(s) with the given id(s)."""
         results = []
-        payload = kwargs
         for oid in order_ids:
-            payload.update({'id': oid})
-            r = self.request('cancel_order/', authenticate=True, params=payload)
+            kwargs.update({'id': oid})
+            r = self.request('cancel_order/', authenticate=True, params=kwargs)
             results.append(r)
         return results if len(results) > 1 else results[0]
 
     @format_with(BitstampFormattedResponse)
-    def wallet(self, *args, **kwargs):
+    def wallet(self, *args, pair=None, **kwargs):
         """Return account's wallet."""
-        if 'pair' in kwargs:
+        endpoint = 'balance/'
+        if pair:
             try:
-                pair = kwargs['pair'].format_for(self.name).lower()
+                pair = pair.format_for(self.name).lower()
             except AttributeError:
-                pair = kwargs['pair']
-
-            return self.request('balance/%s/' % pair, authenticate=True, params=kwargs)
-        return self.request('balance/', authenticate=True, params=kwargs)
+                pair = pair
+            endpoint += pair + '/'
+        return self.request(endpoint, authenticate=True, params=kwargs)
 
     ###########################
     # Exchange Specific Methods
     ###########################
 
     @check_and_format_pair
-    def hourly_ticker(self, pair, **kwargs):
+    def hourly_ticker(self, pair):
```

Comment:
if it's optional, make it a kwarg instead.

Reply:
I know it's optional, I think it's bad design of the bitstamp API. In all the other methods, because they are basic methods, `pair` was added to the method signature, the only missing method to have everything consistent at least in bitstamp to bitex is this method, thus I'd suggest to keep pair as parameter in the signature. Else in all the other methods you do like this:
```
api = bitex.Bitstamp()
btcusd_order_book = api.order_book('btcusd')
btcusd_transactions = api.trades('btcusd')
btc_usd_open_orders = api.open_orders('btcusd')
btcusd_ticker = api.ticker('btcusd')
```

And now the hourly ticker:

```
btcusd_hourly_ticker = api.ticker('btcusd')
```

With kwargs:

```
btcusd_ticker = api.ticker(**{'pair': 'btcusd'})
```

A bit ugly & inconsistent, don't you think?

But it should not contain "if pair", I should rather fix the endpoint to include it directly, like this:

```
    @check_and_format_pair
    def hourly_ticker(self, pair):
        """Return the hourly ticker for the given pair."""
        return self.request('ticker_hour/' + pair)
```

Agree?

Reply:
Actually, checking again the official bitstamp API, v2 **requires** the pair parameter.

https://www.bitstamp.net/api/#ticker-hour